### PR TITLE
TS-4470: ASAN stack-buffer-overflow when slow log is enabled

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/189/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/298/ for details.

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -6890,7 +6890,8 @@ HttpSM::update_stats()
     int offset = 0;
     int skip = 0;
 
-    t_state.hdr_info.client_request.url_print(url_string, sizeof url_string, &offset, &skip);
+    t_state.hdr_info.client_request.url_print(url_string, sizeof(url_string), &offset, &skip);
+    url_string[sizeof(url_string) - 1] = 0; // NULL terminate the string
```
Comment (SolidWallOfCode) [inline proxy/http/HttpSM.cc]: Shouldn't this be `url_string[offset] = 0;`?

Comment (bryancall) [inline proxy/http/HttpSM.cc]: Yeah, that would be better.

Comment (oknet) [inline proxy/http/HttpSM.cc]: The int HTTPHdr::url_print(char *buff, int length, int *offset, int *skip) only called here, can we define the output buffer ( buff ) is a '\0' terminated string?

**Pull request merged**